geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/HARegionQueueStatsCloseRegressionTest.java

```diff
@@ -0,0 +1,191 @@
+import static org.apache.geode.test.dunit.VM.getVM;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.SocketException;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.client.ClientRegionFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.client.Pool;
+import org.apache.geode.cache.client.PoolManager;
+import org.apache.geode.cache.server.CacheServer;
+import org.apache.geode.internal.cache.InternalCacheServer;
+import org.apache.geode.internal.cache.ha.HARegionQueue;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.CacheRule;
+import org.apache.geode.test.dunit.rules.ClientCacheRule;
+import org.apache.geode.test.dunit.rules.DistributedTestRule;
+import org.apache.geode.test.junit.categories.ClientServerTest;
+import org.apache.geode.test.junit.categories.DistributedTest;
+import org.apache.geode.test.junit.rules.serializable.SerializableTestName;
+
+/**
+ * This tests the fix for bug 37210. Reason for the bug was that HARegionQueue's destroy was not
+ * being called on CacheClientProxy's closure. As a result, stats were left open.
+ *
+ * <p>
+ * TRAC #37210: HARegionQueueStats are never closed
+ */
+@Category({DistributedTest.class, ClientServerTest.class})
+public class HARegionQueueStatsCloseRegressionTest implements Serializable {
+
+  private String uniqueName;
+  private String regionName;
+  private String hostName;
+
+  private int port;
+
+  private VM server;
+  private VM client;
+
+  @Rule
+  public DistributedTestRule distributedTestRule = new DistributedTestRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Rule
+  public ClientCacheRule clientCacheRule = new ClientCacheRule();
+
+  @Rule
+  public SerializableTestName testName = new SerializableTestName();
+
+  @Before
+  public void setUp() throws Exception {
+    server = getVM(0);
+    client = getVM(2);
+
+    uniqueName = getClass().getSimpleName() + "_" + testName.getMethodName();
+    regionName = uniqueName + "_region";
+    hostName = getHostName();
+
+    port = server.invoke(() -> createServerCache());
+  }
+
+  /**
+   * This test does the following: <br>
+   * 1) Create the client <br>
+   * 2) Do some operations from the cache-server <br>
+   * 3) Stop the primary cache-server <br>
+   * 4) Explicitly close the CacheClientProxy on the server. <br>
+   * 5) Verify that HARegionQueue stats are closed and entry for the ha region is removed from
+   * dispatchedMessagesMap.
+   */
+  @Test
+  public void testHAStatsCleanup() throws Exception {
```

Comment:
Can we get more of the responsibility into the test name? (I can't think of a concise way to describe the responsibility.)